R/render-table.R

```diff
@@ -3,33 +3,104 @@
-#' @param quoted Is \code{expr} a quoted expression (with \code{quote()})? This
-#'   is useful if you want to save an expression in a variable.
+#' @param quoted Is \code{expr} a quoted expression (with \code{quote()})?
+#'   This is useful if you want to save an expression in a variable.
 #' @param func A function that returns an R object that can be used with
 #'   \code{\link[xtable]{xtable}} (deprecated; use \code{expr} instead).
 #'
 #' @export
-renderTable <- function(expr, ..., env=parent.frame(), quoted=FALSE, func=NULL) {
+renderTable <- function(expr, striped=FALSE, condensed=TRUE,
+                        hover=FALSE, bordered=FALSE,
+                        width="auto", align=NULL,
+                        rownames=FALSE, colnames=TRUE,
+                        digits=NULL, na="NA", ...,
+                        env=parent.frame(), quoted=FALSE, func=NULL) {
   if (!is.null(func)) {
     shinyDeprecated(msg="renderTable: argument 'func' is deprecated. Please use 'expr' instead.")
   } else {
     installExprFunction(expr, "func", env, quoted)
   }
 
+  # A small helper function to create a wrapper for an argument that was
+  # passed to renderTable()
+  createWrapper <- function(arg) {
+    if (is.function(arg)) wrapper <- reactive({ arg() })
+    else wrapper <- function() arg
+    return(wrapper)
+  }
+
+  # Create wrappers for most arguments so that functions can also be passed
+  # in, rather than only literals (useful for shiny apps)
+  stripedWrapper <- createWrapper(striped)
+  condensedWrapper <- createWrapper(condensed)
+  hoverWrapper <- createWrapper(hover)
+  borderedWrapper <- createWrapper(bordered)
+  widthWrapper <- createWrapper(width)
+  alignWrapper <- createWrapper(align)
+  rownamesWrapper <- createWrapper(rownames)
+  colnamesWrapper <- createWrapper(colnames)
+  digitsWrapper <- createWrapper(digits)
+  naWrapper <- createWrapper(na)
+
+  # Main render function
   markRenderFunction(tableOutput, function() {
-    classNames <- getOption('shiny.table.class') %OR% 'data table table-bordered table-condensed'
+    striped <- stripedWrapper()
+    condensed <- condensedWrapper()
+    hover <- hoverWrapper()
+    bordered <- borderedWrapper()
+    format <- c(striped=striped, condensed=condensed,
+                hover=hover, bordered=bordered)
+    width <- widthWrapper()
+    align <- alignWrapper()
+    rownames <- rownamesWrapper()
+    colnames <- colnamesWrapper()
+    digits <- digitsWrapper()
+    na <- naWrapper()
+
+    # For css styling
+    classNames <- paste0("table shiny-table")
+    for (i in seq_len(length(format))) {
+      if (format[i]) {
+        classNames <- paste0( classNames, " table-", names(format)[i])
+      }
+    }
+
     data <- func()
 
-    if (is.null(data) || identical(data, data.frame()))
-      return("")
+    # Return empty string if no data is provided
+    if (is.null(data) || identical(data, data.frame())) return("")
```

Comment:
I see that you're returning empty string because that's what the old code did. That was extremely early code I believe, these days it's more idiomatic shiny to return NULL.